### ISSUE TITLE
Transitions between lots give perception of slowness

### ISSUE THREAD

**Issue by allthesignals** [NYCPlanning/labs-zola#477]
Some transitions between lots lag, either on the click event or because Ember routes do not complete a transition until after the model hook resolves.

![delay](https://user-images.githubusercontent.com/5004319/34440187-60ae1e3a-ec81-11e7-9e08-86d39d185c49.gif)

Next steps:
 - [ ] Add clicked geometry to map, style as "selected"
 - [ ] Replace that geometry with the full geometry once the network request completes
 - [ ] Replace spinner animation with "loading" placeholder block content

**Comment by allthesignals** [comment]
Ember concurrency doesn't cancel on the route transitions.

**Comment by allthesignals** [comment]
This seems to be caused by the model hook, which makes a network request for the lot data. I wonder if this should be cancellable too.

**Comment by allthesignals** [comment]
@andycochran @chriswhong have you noticed this? Is this an issue? If you slow down your network, navigating between lots waits for the data to come back before starting a new transition.

**Comment by allthesignals** [comment]
Set up lot route to use ember concurrency. I think the problem is two things: you expect the lot to highlight when you click it, it doesn't, and the route won't transition unless that data is loaded. We probably need a way to handle the feature highlighting through mapbox.

**Comment by allthesignals** [comment]
In 6a64b03, lot routes return ember concurrency tasks rather than regular promises, which prevent subsequent route transitions on slow networks. Instead, tasks return “restartable” tasks that cancel when the user clicks additional route. The URL transitions, and the template will display a load spinner. However, the lot highlighting does not change because this is pulled from data. We will need to address this in later issues.

**Comment by allthesignals** [comment]
Skeleton screen loading, now possible with tasks: https://emberway.io/skeleton-screen-loading-in-ember-js-2f7ac2384d63

**Comment by allthesignals** [comment]
https://github.com/michalsnik/ember-content-placeholders

**Comment by allthesignals** [comment]
This comment has an interesting way of handling these transitions by stilling displaying previous data (while new data loads):
https://medium.com/@weaver.devin/in-cases-where-you-may-want-to-keep-showing-the-old-data-while-new-data-is-still-loading-you-can-9b96b78105e5

```javascript
// ==== app/routes/blog.js ====
import Route from '@ember/routing/route';
import { task } from 'ember-concurrency';
export default Route.extend({
  model: function (params){
    return {
      previous: this.get('blogTask.lastSuccessful'),
      current: this.get('blogTask').perform(params.blog_id)
    };
  },
  blogTask: task(function *(blogId){
    let blog = yield this.get('store').findRecord('blog', blogId);
    return blog;
  }).cancelOn('deactivate')
});
// ==== app/controllers/blog.js ====
export default Ember.Controller.extend({
  blog: Ember.computed.or('model.{current,previous}.value'),
  isLoading: Ember.computed.reads('model.current.isRunning'),
  error: Ember.computed.reads('model.current.error')
});
```

**Comment by evillagemelissa** [comment]
I'm not sure if this relates... but if I use ZoLa, type an address, I get the map and sidebar for that address. If I type in a new address, the sidebar with new information repopulates, but the map doesn't repopulate and stays blank. Even clicking on the ZoLa header or opening a new tab to refresh and start over doesn't work.

**Comment by evillagemelissa** [comment]
ZoLa not populating the map consistently (in Firefox) has led me to use old ZoLa so I can get questions answered efficiently for Zoning Help Desk

**Comment by allthesignals** [comment]
@evillagemelissa can you let me know if this has improved? does map populate consistently? If not, what are the steps you take that lead to this behavior? thanks!

**Comment by hannahkates** [comment]
Duplicate of #681. I'll reference this issue in 681 to provide additional context
